dizoo/metadrive/env/drive_wrapper.py

```diff
@@ -0,0 +1,114 @@
+import copy
+import numpy as np
+from ding.envs.env.base_env import BaseEnvTimestep
+from ding.envs.common.env_element import EnvElementInfo
+from ding.torch_utils.data_helper import to_ndarray
+from dizoo.metadrive.env.drive_utils import BaseDriveEnv, deep_merge_dicts
+
+
+class DriveEnvWrapper(gym.Wrapper):
+    """
+    Environment wrapper to make ``gym.Env`` align with DI-engine definitions, so as to use utilities in DI-engine.
+    It changes ``step``, ``reset`` and ``info`` method of ``gym.Env``, while others are straightly delivered.
+
+    :Arguments:
+        - env (BaseDriveEnv): The environment to be wrapped.
+        - cfg (Dict): Config dict.
+
+    :Interfaces: reset, step, info, render, seed, close
+    """
+
+    config = dict()
+
+    def __init__(self, env: BaseDriveEnv, cfg: Dict = None, **kwargs) -> None:
+        if cfg is None:
+            self._cfg = self.__class__.default_config()
+        elif 'cfg_type' not in cfg:
+            self._cfg = self.__class__.default_config()
+            self._cfg = deep_merge_dicts(self._cfg, cfg)
+        else:
+            self._cfg = cfg
+        self.env = env
+        if not hasattr(self.env, 'reward_space'):
+            self.reward_space = gym.spaces.Box(low=-float('inf'), high=float('inf'), shape=(1,))
+        self.action_space = self.env.action_space
+        self.env = env
+
+    def reset(self, *args, **kwargs) -> Any:
+        """
+        Wrapper of ``reset`` method in env. The observations are converted to ``np.ndarray`` and final reward
+        are recorded.
+
+        :Returns:
+            Any: Observations from environment
+        """
+        obs = self.env.reset(*args, **kwargs)
+        obs = to_ndarray(obs, dtype=np.float32)
+        if isinstance(obs, np.ndarray) and len(obs.shape) == 3:
+            obs = obs.transpose((2, 0, 1))
+        elif isinstance(obs, dict):
+            vehicle_state = obs['vehicle_state']
+            birdview = obs['birdview'].transpose((2,0,1))
+            obs = {'vehicle_state': vehicle_state, 'birdview': birdview}
+        self._final_eval_reward = 0.0
+        self._arrive_dest = False
+        return obs
+
+    def step(self, action: Any = None) -> BaseEnvTimestep:
+        """
+        Wrapper of ``step`` method in env. This aims to convert the returns of ``gym.Env`` step method into
+        that of ``ding.envs.BaseEnv``, from ``(obs, reward, done, info)`` tuple to a ``BaseEnvTimestep``
+        namedtuple defined in DI-engine. It will also convert actions, observations and reward into
+        ``np.ndarray``, and check legality if action contains control signal.
+
+        :Arguments:
+            - action (Any, optional): Actions sent to env. Defaults to None.
+
+        :Returns:
+            BaseEnvTimestep: DI-engine format of env step returns.
+        """
+        action = to_ndarray(action)
+        obs, rew, done, info = self.env.step(action)
+        #draw_multi_channels_top_down_observation(obs, show_time=4.5)
```

Comment:
What is the function of this line. If necessary, add an option to choose whether to use this particular obs.